File: lib/whooks_web/plugs/global_filters.ex
defmodule WhooksWeb.Plugs.GlobalFilters do
  import Plug.Conn
  import Inertia.Controller

  require Logger

  def init(default), do: default

  def call(%Plug.Conn{params: params} = conn, _default) do
    last = Map.get(params, "last", "1h")
    interval = Map.get(params, "interval", "minute")

    Logger.info("[Plugs.GlobalFilters] last: #{params["last"]}, interval: #{params["interval"]}")

    conn
    |> assign(:global_filters, %{last: last, interval: interval})
    |> assign_prop(:global_filters, %{last: last, interval: interval})
  end
end

File: assets/js/types/globalFilters.ts
export type Interval =
  | "1s"
  | "15s"
  | "30s"
  | "45s"
  | "1m"
  | "15m"
  | "30m"
  | "45m"
  | "1h"
  | "1d"
  | "1w"
  | "1mo"
  | "minute"
  | "hour"
  | "day";

export type Last = "1m" | "1h" | "12h" | "24h" | "48h" | "1w" | "1mo";

export type GlobalFilters = { last: Last; interval: Interval };